sound/soc/sof/ipc4-topology.c

```diff
@@ -1385,9 +1388,18 @@ static int snd_sof_get_nhlt_endpoint_data(struct snd_sof_dev *sdev, struct snd_s
 	switch (linktype) {
 	case SOF_DAI_INTEL_DMIC:
 		nhlt_type = NHLT_LINK_DMIC;
-		bit_depth = params_width(params);
 		channel_count = params_channels(params);
 		sample_rate = params_rate(params);
+		bit_depth = params_width(params);
+		/*
+		 * Look for 32-bit blob first instead of 16-bit if copier
```

Comment:
@ujfalusi I feel a bit uncomfortable with clubbing the topology formats with the NHLT blob detection. If the NHLT blobs came from topology always, this would make sense but when the blobs are from the BIOS, why do we need to worry about what the topology contains at all?

Reply:
@ranj063, if the copier supports single bit depth (defined by topology) then we can only load an NHLT blob which also supports this and only this bit depth, no?
It does not matter where the NHLT is coming from, the copier config and the blob must be compatible, so we take this restriction from the copier's format support and use it to look for the blob.

I think it is more pragmatic than using 32bit blob unconditionally even if the copier only supports 16bit.

Reply:
@ranj063 you really need to chime in, we can't have two PRs suggested for the same purpose...